Client/Shared/NavMenu.razor

```diff
@@ -18,94 +19,277 @@
+                                    <li class="nav-item px-3">
+                                        <ContextMenuTrigger MenuId="@SidebarContextMenu.GetMenuId(article.Id, SidebarContextMenu.SidebarContextMenuType.ARTICLE)">
+                                            <NavLink class="nav-link" id="@($"nav-link-article-{article.Id}")" href="@($"/Section/{section.Id}/Article/{article.Id}")">
+                                                <span class="oi oi-justify-left" aria-hidden="true"></span> @article.Title
+                                            </NavLink>
+                                        </ContextMenuTrigger>
+                                        <SidebarContextMenu Article="@article" OnDelete="@OnArticleDelete" Type="@SidebarContextMenu.SidebarContextMenuType.ARTICLE" ShowEdit="@false"/>
+                                    </li>
+                                }
                                 <li class="nav-item px-3">
-                                    <NavLink class="nav-link" href="@article.Id">
-                                        <span class="oi oi-justify-left" aria-hidden="true"></span> @article.Title
-                                    </NavLink>
+                                    <button type="button" class="btn btn-success w-100" @onclick="@(() => Create(section.Id))">
+                                        Create Article&nbsp;<i class="oi oi-plus"></i>
+                                    </button>
                                 </li>
-                            }
-                        </ul>
+                            </ul>
+                        }
                     }
-                }
-            </li>
+                </li>
+            }
+            @if (_showSectionCreate)
+            {
+                <li class="nav-item px-3">
+                    <SidebarEditSection
+                        SectionId="@(-1)"
+                        InitialTitle=""
+                        OnSave="@OnSectionCreateConfirm"/>
+                </li>
+            }
         }
     </ul>
 </div>
 
 @code {
     [Inject]
-    private NavigationManager _navigationManager { get; set; }
-    private SectionService    _sectionService    { get; set; }
-    private ArticleService    _articleService    { get; set; }
+    private NavigationManager NavigationManager { get; set; }
+    [Inject]
+    private IJSRuntime JsRuntime { get; set; }
+
+    private SectionService    _sectionService;
+    private ArticleService    _articleService;
     private bool collapseNavMenu = true;
     private string NavMenuCssClass => collapseNavMenu ? "collapse" : null;
 
-    private long             expandedSectionId       { get; set; }
-    private List<ArticleDto> expandedSectionArticles { get; set; }
-    private bool             isLoadingArticles       { get; set; }
+    private bool _isLoadingAll = false;
+    private long _editingSectionId = -1;
+    private bool _showSectionCreate = false;

```

Comment:
minor, looks like you were aligning some things and not others.

Reply:
yeah, the formatter in Rider doesn't seem to understand that the `.razor` extension contains C# code 😬